The Sawtooth Private Transaction Families lab

sgurary opened the pull request.

tkuhrt commented on labs/PrivateLedger.md:
```diff
@@ -0,0 +1,19 @@
+ 
+
+# Scope of Lab
+This hyperledger lab project scope is to develop and test the creation of a private ledger application, that works with Intel's SGX technology.
```
hyperledger lab project -> Hyperledger lab

tkuhrt replied:
Will this lab be limited in scope to Hyperledger Sawtooth? Could you add a statement about which projects this lab will work with?

tkuhrt commented on labs/PrivateLedger.md:
```diff
@@ -0,0 +1,19 @@
+shefy.gurary@intel.com, yoni.wolf@intel.com, oron.lenz@intel.com, ronen.shem-tov@intel.com, idan.sorski@intel.com (others might be added later on)
+
+# Sponsor
+shefy.gurary@intel.com 
```
A sponsor must be a maintainer of one of the Hyperledger projects, a TSC member, or a WG chair.

tkuhrt commented:
Can you rename the .md lab to private-ledger.md to reflect naming convention for Github repository?

tkuhrt commented on labs/private-ledger.md:
```diff
@@ -8,7 +8,7 @@ The purpose of the software we develop (“Private Ledger”) is to provide a me
 
 # Scope of Lab
-This hyperledger lab project scope is to develop and test the creation of a private ledger application, that works with Intel's SGX technology.
+This Hyperledger lab project scope is to develop and test the creation of a private ledger application, that works with Intel's SGX technology.
```
remove the word project. Labs are not to be confused with Hyperledger projects.

tkuhrt commented:
@sgurary : Looks like you need to amend the last commit to ensure the -s option is specified.

tkuhrt commented on labs/private-ledger.md:
```diff
@@ -0,0 +1,19 @@
+shefy.gurary@intel.com, yoni.wolf@intel.com, oron.lenz@intel.com, ronen.shem-tov@intel.com, idan.sorski@intel.com (others might be added later on)
+
+# Sponsor
+shefy.gurary@intel.com 
```
Still need a sponsor that is a maintainer of one of the Hyperledger projects, a TSC member, or a WG chair.

sgurary commented:
Hi,

- I updated the proposal not to include the project word. (As you said - labs are not to be confused with projects)
- I added a sponsor that fits the criteria 
- I did the last commit with the -s flag (I did the last few previous commits from the Github site, what caused the issue)

Thanks, Shefy

tkuhrt commented:
Looks like the -s option still has not taken. You will need to squash your merge into a single commit with the -s option.

tkuhrt commented on labs/private-ledger.md:
```diff
@@ -0,0 +1,20 @@
+This Hyperledger lab scope is to develop and test the creation of a private ledger application, that works with Intel's SGX technology.
+
+# Initial Committers
+shefy.gurary@intel.com, yoni.wolf@intel.com, oron.lenz@intel.com, ronen.shem-tov@intel.com, idan.sorski@intel.com (others might be added later on)
```
Can you provide the github IDs in addition to the email addresses for the committers. This will help us when we go to create the repository making sure that each person is added to the lab.

lehors commented:
@sgurary The link on "Details" (https://github.com/probot/dco#how-it-works) has the explanation.

sgurary commented:
I added the line and the flag for the signed off, but the commit still fails.
@lehors any idea?

tkuhrt commented:
@sgurary : It looks like some of the commits have the signoff, but not all of them. Can you squash your commits into a single commit with a signoff?

tkuhrt commented:
I found some instructions that might help:
1. If you need to re-sign a bunch of previous commits at once, find the earliest unsigned commit using `git log --show-signature` and use the HASH of the commit before the earliest unsigned commit in this command: `git rebase --exec 'git commit --amend --no-edit -n -s' -i HASH`. This will sign every commit from most recent to right before the HASH.
1. You will probably need to do a force push `git push -f` if you had previously pushed unsigned commits to a remote.

sgurary commented:
Thanks @tkuhrt 

I acted according to your instructions. 
It seems the process has performed as expected.

Can you check and tell me, if there are additional fixes we need to do?

Thanks, Shefy

tkuhrt commented on design/temp.txt:
```diff
@@ -0,0 +1 @@
+Nothing at the moment
```
Need to remove this file. Design for the lab will be put into the lab repo once created.

tkuhrt commented:
@sgurary : Thanks for the update. It is strange that the DCO is expecting a noreply email. If the lab stewards agree that this lab makes sense (@lehors @trbs @vipinsun @ryjones), we should try to override the DCO bot and merge this into labs.

You will need to remove the design/temp.txt file before we do this though.

Stewards, what say you?

trbs commented:
Think this is an interesting proposal and in line with the labs. 

@sgurary could you clarify if this proposal is to build a new blockchain/ledger ? Or just the component that facilitates doing private transactions based on SGX ?

If the later I would like to think about another name since Private Ledger to me indicates a (complete) distributed ledger with focus on private state.

lehors commented:
I agree with @trbs, the title and description should be clearer as to what is to be developed and whether it depends on a specific platform (sawtooth?)
My colleague Marcus Brandenburger presented at the Amsterdam hackfest similar work around Fabric. It would be good to allow all of these projects to co-exist without creating confusion. Making the name and description explicit will help achieve that.
Thanks Tracy for shepherding this PR so far.

sgurary commented:
Hi,
@tkuhrt - I removed the file as requested.
@trbs, @lehors - This proposal is indeed for a new blockchain/ledger which is private. We use the same name internally for this project.  

Thanks for the fast, and helpful feedback, Shefy

tkuhrt commented:
@sgurary : I still see the `design/temp.txt` in the commits. Have you pushed your changes to remove this file? Also, I noticed in the proposal, that there are github IDs for two of the initial committers. Can we get the Github IDs for the other committers added to the proposal?

sgurary commented:
@tkuhrt  - I deleted temp.txt and committed directly here. 
About the committers from the proposal. They are probably not listed in Github yet. Should I remove them?

ronenshemtov commented:
@tkuhrt, @trbs, @lehors . Thanks for your feedback. To answer your question:  This proposal is to build the component that facilitates private transactions on Sawtooth hyperledger based on SGX. We plan to expand it to Fabric and potentially other hyperledgers in the future.
As for the project name, does the following name makes a better sense to you? Private Ledger on Hyperledger Sawtooth.

lehors commented:
Sorry but I'm having trouble reconciling @ronenshemtov's answer with @sgurary's:

> This proposal is to build the component that facilitates private transactions on Sawtooth hyperledger based on SGX. We plan to expand it to Fabric and potentially other hyperledgers in the future.

vs

> This proposal is indeed for a new blockchain/ledger which is private.

sgurary commented:
I guess this is a bit because I and Ronen understood the question differently.

The question was:

Could you clarify if this proposal is to build a new blockchain/ledger ? Or just the component that facilitates doing private transactions based on SGX ?
 
We're not developing a new type of ledger if that's the question. We're working with Sawtooth. But I'm also not sure if you could consider it as a component. I think Ronen is more qualified from me to answer this question, but we do need more clarifications on what are the implications on regarding this project as a component. I do think this project should be a stand-alone lab if that's the question. 

Anyway, I do agree that the name should probably change to reflect the use of sawtooth and SGX.

lehors commented:
The question is really not about whether this proposal should be allowed as a lab. I don't think anyone is objecting to it. It is merely a question of making the title and description clear enough so that people looking into it without inside knowledge could get a fairly good understanding of what the lab is about.

tkuhrt commented:
We recently created [fabric-secure-chaincode](https://github.com/hyperledger-labs/fabric-secure-chaincode) lab. Maybe we should name this one sawtooth-secure-transaction-families to make it clear the purpose of this lab is
  1. to work with Hyperledger Sawtooth
  2. to make the transactions families more secure by utilizing the Intel SGX chip

This would include three changes:
  1. rename private-ledger.md to sawtooth-secure-transaction-families.md
  2. Update the lab name (and other references to private ledger) in the MD file
  3. Update the pull request title to reflect the name change

tkuhrt commented:
Also, it looks like the last commit that was done missed the DCO sign off. We will now need to fix that. Since the last commit may have been done directly in the Github UI, we may need a separate PR to fix this. Before we get there, let's make sure that everyone is good with updates.

ronenshemtov commented:
Thanks for the prompt replies.
I’ll check the name with the team and update by EOW.

Regards,
Ronen Shem-Tov
Platfrom Security Division (PSD)
Intel Corporation


From: Tracy Kuhrt [mailto:notifications@github.com]
Sent: Monday, August 27, 2018 21:38
To: hyperledger-labs/hyperledger-labs.github.io <hyperledger-labs.github.io@noreply.github.com>
Cc: Shem-tov, Ronen <ronen.shem-tov@intel.com>; Mention <mention@noreply.github.com>
Subject: Re: [hyperledger-labs/hyperledger-labs.github.io] The Private Ledger project (#13)


We recently created fabric-secure-chaincode<https://github.com/hyperledger-labs/fabric-secure-chaincode> lab. Maybe we should name this one sawtooth-secure-transaction-families to make it clear the purpose of this lab is

  1.  to work with Hyperledger Sawtooth
  2.  to make the transactions families more secure by utilizing the Intel SGX chip

This would include three changes:

  1.  rename private-ledger.md to sawtooth-secure-transaction-families.md
  2.  Update the lab name (and other references to private ledger) in the MD file
  3.  Update the pull request title to reflect the name change

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/hyperledger-labs/hyperledger-labs.github.io/pull/13#issuecomment-416325159>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AoQNaWh0HvOhkkPZAbNwN7M2RbcLvRRyks5uVDyRgaJpZM4ScFP_>.
---------------------------------------------------------------------
Intel Israel (74) Limited

This e-mail and any attachments may contain confidential material for
the sole use of the intended recipient(s). Any review or distribution
by others is strictly prohibited. If you are not the intended
recipient, please contact the sender and delete all copies.

tkuhrt commented:
@ronenshemtov : The reason I made transaction families plural was to remove any possible confusion that this was a transaction family implementation. Other than that, I do not have a problem with the suggested name.

@trbs and @lehors can you provide your thoughts as to whether the name change resolves your concerns from above?

lehors commented:
Yes, it's fine with me.
Thanks.

ronenshemtov commented:
Makes sense so final name is: "sawtooth-private-transaction-families"
We'll  ping back once we make the needed updates (next week)

Thanks

sgurary commented:
> Based on the feedback we'd like to call our project : sawtooth-private-transaction-family
> Please confirm and we'll make the following updates:
> 
> 1. rename private-ledger.md to sawtooth-private-transaction-familiy.md
> 2. Update the lab name (and other references to private ledger) in the MD file
> 3. Update the pull request title to reflect the name change

@tkuhrt @ronenshemtov 
Done

tkuhrt commented:
@sgurary : Looks like we still have DCO issues with this pull request and the design/temp.txt file needs to be removed.

sgurary commented:
@tkuhrt 
I removed the file, How should we solve the DCO issues?

tkuhrt commented:
Check out step #10 in [this gist](https://gist.github.com/tkuhrt/10211ae0a26a91a8c030d00344f7d11b)

sgurary commented:
I did - didn't seem to help. And it continues to go back to the old proposal file

tkuhrt commented:
Someone else was having problems on a different branch. He did not realize that all commits had to have the DCO sign-off. So you would need to squash all your commits into just one. If you are having trouble, it might be time to close this PR and start with a fresh one that only includes the latest .md file and has one commit with one sign-off.

tkuhrt commented:
Before we go too far, I remembered that the DCO now provides details on fixing this issue. See https://github.com/hyperledger-labs/hyperledger-labs.github.io/pull/13/checks?check_run_id=17651183 for specifics on how to fix the DCO signoff.

sgurary commented:
I did the action required, but still the check failed, any idea?

sgurary commented:
Ok - starting fresh according to your suggestions.

Thanks, Shefy
